src/main/java/io/cdap/plugin/salesforce/plugin/source/batch/SalesforceBulkRecordReader.java

```diff
@@ -214,7 +247,10 @@ private String[] waitForBatchResults(BulkConnection bulkConnection)
       }
 
       if (info.getState() == BatchStateEnum.Completed) {
-        return bulkConnection.getQueryResultList(jobId, batchId).getResult();
+        Failsafe.with(SalesforceSplitUtil.getRetryPolicy(initialRetryDuration, maxRetryDuration, maxRetryCount))
+          .run(() -> batchInfoList[0] = getQueryResultList(bulkConnection));
+        return batchInfoList[0];
```

Comment:
what is the use of storing `batchInfoList` and then returning?

```
return Failsafe.with(SalesforceSplitUtil.getRetryPolicy(initialRetryDuration, maxRetryDuration, maxRetryCount))
          .run(() -> getQueryResultList(bulkConnection));
```

Reply:
This does not make sense, how does storing it in the variable help here?

Can you share an example explaining this?